### PR TITLE
Add Config object to remove session variables

### PR DESCRIPTION
**Pull request by Nispeon**
Added an object that stores all the necessary data to create a Secret Santa, and passed it in session instead of all the variables that we used to.
Also updated the SecretSanta object with the same idea, replaced a bunch of parameters (not all) by one single $config parameter containing most of the data needed.

Tests and QA should be taken care of as well

### CODE REVIEW

File: src/Controller/SantaController.php [hunk 1]
```diff
@@ -155,57 +155,49 @@ public function message(Rudolph $rudolph, FormFactoryInterface $formFactory, Req
...
-            $session->set('message', $message);
-            foreach ($form->getData() as $data => $note) {
-                if (str_contains($data, 'notes-')) {
-                    $notes[str_replace('notes-', '', $data)] = $note;
-                }
-            }
-            $session->set('notes', $notes);
+            $session->set('config', $config);
+
+            $errors = array_map(function (FormError $error) {
```
Comment (pyrech) [inline src/Controller/SantaController.php:160]: could you move the errors handling after the block `if ($form->isValid()) {`? We do not need it if the form is valid :wink:

File: src/Controller/SantaController.php [hunk 2]
```diff
@@ -224,49 +216,47 @@ public function sendSampleMessage(Request $request, string $application): Respon
...
+        $form->handleRequest($request);
+
+        $notes = $config->getNotes();
 
-        $candidates = array_filter($notes ? array_keys($notes) : $selectedUsers, function ($id) use ($application) {
+        $candidates = array_filter($notes ? array_keys($notes) : $config->getSelectedUsers(), function ($id) use ($application) {
             return $application->getAdmin()->getIdentifier() !== $id;
         });
 
         $receiver = $candidates ? $candidates[array_rand($candidates)] : $application->getAdmin()->getIdentifier();
```
Comment (pyrech) [inline src/Controller/SantaController.php:239]: Could you move the notes & candidates inside the block `$form->isSubmitted()`?

File: src/Slack/MessageSender.php [hunk 3]
```diff
@@ -58,6 +58,7 @@ public function sendSecretMessage(SecretSanta $secretSanta, string $giver, strin
         ];
 
         $receiverUser = $secretSanta->getUser($receiver);
+        dump($receiver, $secretSanta, $receiverUser);
```
Comment (pyrech) [inline src/Slack/MessageSender.php]: dump to remove

**Review by pyrech — CHANGES_REQUESTED** [review]

File: src/Controller/SantaController.php (hunk 1 above)
Comment (Nispeon) [reply src/Controller/SantaController.php]: You're right, fixed 😊

File: src/Controller/SantaController.php (hunk 2 above)
Comment (Nispeon) [reply src/Controller/SantaController.php]: 👌

File: templates/santa/participants.html.twig [hunk 4]
```diff
@@ -62,7 +62,8 @@
                                                 <div class="no-result is-hidden" id="no-user-matching">
                                                     No user matching.
                                                 </div>
-                                                {% for userForm in form.users %}
+{#                                                {{ dump(form.selectedUsers|first) }}#}
```
Comment (pyrech) [inline templates/santa/participants.html.twig:65]: dump to remove

File: src/Controller/SantaController.php [hunk 5]
```diff
@@ -137,15 +125,27 @@ public function message(Rudolph $rudolph, FormFactoryInterface $formFactory, Req
...
-
-        $builder = $formFactory->createBuilder(MessageType::class, ['message' => $message], [
-            'selected-users' => $selectedUsers,
+
+        /** @var Config $config * */
+        $config = $session->get('config');
+
+        $selectedUsersAsArray = $config->getSelectedUsers();
+
+        $notes = array_filter($config->getNotes(), function ($userIdentifier) use ($selectedUsersAsArray) {
```
Comment (pyrech) [inline src/Controller/SantaController.php:137]: Could you add a comment to explain that we clean/prepare the notes according to the selected users here

**Review by pyrech — CHANGES_REQUESTED** [review]

File: src/Controller/SantaController.php (hunk 5 above)
Comment (Nispeon) [reply src/Controller/SantaController.php:138]: Added ^^

**Review by pyrech — APPROVED** [review]

**Pull request merged**